projects/cds/src/merchandising/facade/cds-merchandising-product.service.ts

```diff
@@ -18,31 +33,74 @@ export class CdsMerchandisingProductService {
     protected strategyConnector: MerchandisingStrategyConnector,
     protected baseSiteService: BaseSiteService,
     protected languageService: LanguageService,
-    private routingService: RoutingService
+    private routingService: RoutingService,
+    protected currencyService: CurrencyService,
+    private productSearchService: ProductSearchService,
+    private converterService: ConverterService
   ) {}
 
+  protected readonly RELEVANCE_CATEGORY = ':relevance:category:';
+  protected readonly RELEVANCE_BRAND = ':relevance:brand:';
+  protected defaultPageSize = 10;
+
   loadProductsForStrategy(
     strategyId: string,
     numberToDisplay?: number
   ): Observable<MerchandisingProducts> {
     return combineLatest([
       this.baseSiteService.getActive(),
       this.languageService.getActive(),
-      this.routingService
-        .getRouterState()
-        .pipe(map(state => state.state.params['productCode'])),
+      this.routingService.getRouterState(),
+      this.getFacets(),
     ]).pipe(
-      map(([site, language, productId]: [string, string, string]) => {
-        const strategyRequest: StrategyRequest = {
-          site,
-          language,
-          pageSize: numberToDisplay,
-          productId,
-        };
-        return strategyRequest;
-      }),
-      switchMap(context =>
-        this.strategyConnector.loadProductsForStrategy(strategyId, context)
+      map(
+        ([site, language, routerState, facets]: [
+          string,
+          string,
+          RouterState,
+          string
+        ]) => {
+          const productId = routerState.state.params['productCode'];
+          const category = routerState.state.params['categoryCode'];
+          const strategyRequest: StrategyRequest = {
+            site,
+            language,
+            pageSize: numberToDisplay,
+            productId,
+            category,
+            facets,
+          };
+          return strategyRequest;
+        }
+      ),
+      tap(request => console.log('the request we are using - ', request)),
```

Comment:
I know this is not a polished ticket, but just a reminder to remove this console.log

Reply:
Removed